crates/services/gas_price_service/src/lib.rs

```diff
@@ -79,7 +79,7 @@ where
 }
 
 #[derive(Debug, Default)]
-pub struct SharedGasPriceAlgo<A>(Arc<RwLock<Option<A>>>);
+pub struct SharedGasPriceAlgo<A>(Arc<RwLock<A>>);
```

Comment:
Nit (unrelated to this PR): I found it a bit confusing to read this struct, since the declaration has nothing to do with the fact that the content is a gas price algorithm. Since the type is very generic, I'd find it easier to read if it was just named `Shared`.